File: 3159. Find Occurrences of an Element in an Array.py
class Solution:
    def occurrencesOfElement(self, nums: List[int], queries: List[int], x: int) -> List[int]:
        answer = [-1]*len(queries)
        positions = []
        for i in range(len(nums)):
            if nums[i] == x:
                positions.append(i)
        for i in range(len(queries)):
            if queries[i] <= len(positions):
                answer[i] = positions[queries[i]-1]
        return answer

File: 3159. Find Occurrences of an Element in an Array.java
class Solution {
    public int[] occurrencesOfElement(int[] nums, int[] queries, int x) {
        int[] answer = new int[queries.length];
        List<Integer> positions = new ArrayList<>();
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == x) positions.add(i);
        }
        for (int i = 0; i < answer.length; i++) {
            if (queries[i] > positions.size()) answer[i] = -1;
            else answer[i] = positions.get(queries[i]-1);
        }
        return answer;
    }
}